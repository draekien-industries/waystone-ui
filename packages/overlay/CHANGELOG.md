# @waystone/overlay

## 1.0.0

### Major Changes

- bd9dbe7: Major refactor of package structure and component implementation

  ## Migration steps

  Uninstall the old version of `@waystone/*`, then run one of the below scripts depending on the package manager you use:

  | Package manager | Script                                                         |
  | --------------- | -------------------------------------------------------------- |
  | npm             | `npm i @waystone/ui@latest @emotion/react@11 theme-ui@0.6`     |
  | yarn            | `yarn add @waystone/ui@latest @emotion/react@11 theme-ui@0.16` |
  | pnpm            | `pnpm i @waystone/ui@latest @emotion/react@11 theme-ui@0.16`   |

  Replace imports with an import of `@waystone/ui`

  ```diff
  - import { Waystone } from '@waystone/theme';
  - import { Card, Text, Button } from '@waystone/components';
  + import { Waystone, Card, Text, Button } from '@waystone/ui';
  ```
