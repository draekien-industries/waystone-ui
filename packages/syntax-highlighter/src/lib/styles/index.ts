export * from './a11yDark.generated';
export * from './a11yLight.generated';
export * from './agate.generated';
export * from './anOldHope.generated';
export * from './androidstudio.generated';
export * from './arduinoLight.generated';
export * from './arta.generated';
export * from './ascetic.generated';
export * from './atomOneDarkReasonable.generated';
export * from './atomOneDark.generated';
export * from './atomOneLight.generated';
export * from './brownPaper.generated';
export * from './codepenEmbed.generated';
export * from './colorBrewer.generated';
export * from './dark.generated';
export * from './defaultStyle.generated';
export * from './devibeans.generated';
export * from './docco.generated';
export * from './far.generated';
export * from './felipec.generated';
export * from './foundation.generated';
export * from './githubDarkDimmed.generated';
export * from './githubDark.generated';
export * from './github.generated';
export * from './gml.generated';
export * from './googlecode.generated';
export * from './gradientDark.generated';
export * from './gradientLight.generated';
export * from './grayscale.generated';
export * from './hybrid.generated';
export * from './idea.generated';
export * from './intellijLight.generated';
export * from './irBlack.generated';
export * from './isblEditorDark.generated';
export * from './isblEditorLight.generated';
export * from './kimbieDark.generated';
export * from './kimbieLight.generated';
export * from './lightfair.generated';
export * from './lioshi.generated';
export * from './magula.generated';
export * from './monoBlue.generated';
export * from './monokaiSublime.generated';
export * from './monokai.generated';
export * from './nightOwl.generated';
export * from './nnfxDark.generated';
export * from './nnfxLight.generated';
export * from './nord.generated';
export * from './obsidian.generated';
export * from './pandaSyntaxDark.generated';
export * from './pandaSyntaxLight.generated';
export * from './paraisoDark.generated';
export * from './paraisoLight.generated';
export * from './pojoaque.generated';
export * from './purebasic.generated';
export * from './qtcreatorDark.generated';
export * from './qtcreatorLight.generated';
export * from './rainbow.generated';
export * from './routeros.generated';
export * from './schoolBook.generated';
export * from './shadesOfPurple.generated';
export * from './srcery.generated';
export * from './stackoverflowDark.generated';
export * from './stackoverflowLight.generated';
export * from './sunburst.generated';
export * from './tokyoNightDark.generated';
export * from './tokyoNightLight.generated';
export * from './tomorrowNightBlue.generated';
export * from './tomorrowNightBright.generated';
export * from './vs.generated';
export * from './vs2015.generated';
export * from './xcode.generated';
export * from './xt256.generated';
