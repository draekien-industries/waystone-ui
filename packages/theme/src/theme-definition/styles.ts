export const styles = {
  shadows: {
    none: 'none',
    small:
      'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 0 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
    medium:
      'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 4px 2px hsla(0, 0%, 0%, 0.05), 0 0.6px 0.8px hsla(0, 0%, 0%, 0.02), 0 1.8px 3px hsla(0, 0%, 0%, 0.045), 0 7px 12px hsla(0, 0%, 0%, 0.09)',
    large:
      'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 8px 4px hsla(0, 0%, 0%, 0.05), 0 1.2px 1.6px hsla(0, 0%, 0%, 0.02), 0 3.6px 6px hsla(0, 0%, 0%, 0.045), 0 14px 24px hsla(0, 0%, 0%, 0.09)',
  },
  sizes: {
    'extra-small': '1em',
    small: '5em',
    medium: '10em',
    large: '15em',
    'extra-large': '20em',
    half: '50%',
    full: '100%',
  },
} as const;
