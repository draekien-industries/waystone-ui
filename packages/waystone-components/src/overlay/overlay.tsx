/** @jsxImportSource theme-ui */

import { alpha } from '@theme-ui/color';

export interface OverlayProps {
  children: React.ReactNode;
  fullScreen?: boolean;
}

export const Overlay = ({ children, fullScreen = false }: OverlayProps) => (
  <div
    sx={{
      position: fullScreen ? 'fixed' : 'absolute',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%',
      zIndex: 'highest',
      bg: alpha('muted', 0.2),
      backdropFilter: 'blur(2px)',
      pointerEvents: 'none',
    }}
  >
    <div sx={{ opacity: 1 }}>{children}</div>
  </div>
);
