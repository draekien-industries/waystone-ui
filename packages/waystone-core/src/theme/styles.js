export const styles = {
    shadows: {
        sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
        md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
        lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
        'sm-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.125em 0.25em -0.125em #0b0b0bA0',
        'md-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.5em 1em -0.5em #0b0b0bA0',
        'lg-dark': '0 0 0.5em 0 #1b1b1bFF, 0 1em 2em -1em #0b0b0bA0',
    },
    sizes: {
        xs: '1em',
        sm: '5em',
        md: '10em',
        lg: '15em',
        xl: '20em',
        half: '50%',
        full: '100%',
    },
};
export default styles;
