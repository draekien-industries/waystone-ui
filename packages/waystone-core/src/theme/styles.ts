export const styles = {
  shadows: {
    sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
    md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
    lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
    'sm-dark':
      '0 0 0.5em 0 rgba(27, 27, 27, 1), 0 0.5 1em -0.5em rgba(11, 11, 11, 0.73)',
    'md-dark':
      '0 0 1em 0 rgba(27, 27, 27, 1), 0 1em 2em -1em rgba(11, 11, 11, 0.73)',
    'lg-dark':
      '0 0 2em 0 rgba(27, 27, 27, 1), 0 2em 4em -2em rgba(11, 11, 11, 0.73)',
  },
  sizes: {
    xs: '1em',
    sm: '5em',
    md: '10em',
    lg: '15em',
    xl: '20em',
    half: '50%',
    full: '100%',
  },
};

export default styles;
