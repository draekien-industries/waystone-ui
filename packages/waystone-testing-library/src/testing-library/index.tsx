'use client';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render } from '@testing-library/react';
import { Waystone } from '@waystone/theme-provider';
import type { PropsWithChildren } from 'react';
import type React from 'react';
import { useEffect } from 'react';
import { useColorMode } from 'theme-ui';

const TestProvider = ({ children }: PropsWithChildren) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper = ({ children }: PropsWithChildren) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: RenderOptions
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: options?.wrapper || TestProvider, ...options });

export * from '@testing-library/react';
export * from '@testing-library/user-event';

export { DarkModeWrapper, themedRender as render };
