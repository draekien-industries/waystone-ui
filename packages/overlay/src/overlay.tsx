import type { PropsWithChildren } from 'react';
import { Box } from 'theme-ui';
import type { ThemeColor } from '@waystone/theme';
import { absoluteCss, fullScreenCss } from './overlay.styles';
import type { OverlayProps } from './overlay.types';

export const FullScreenOverlay = ({
  children,
  ...rest
}: PropsWithChildren<OverlayProps>) => (
  <Box sx={fullScreenCss(rest)}>
    <Box sx={{ opacity: 1, position: 'relative' }}>{children}</Box>
  </Box>
);

export const AbsoluteOverlay = ({
  children,
  ...rest
}: PropsWithChildren<OverlayProps>) => (
  <Box sx={absoluteCss(rest)}>{children}</Box>
);
export const Overlay = {
  FullScreen: FullScreenOverlay,
  Absolute: AbsoluteOverlay,
} as const;
