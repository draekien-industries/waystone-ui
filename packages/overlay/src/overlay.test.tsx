import { render } from '@waystone/testing-library';
import { Overlay } from './overlay';
import { absoluteCss, fullScreenCss } from './overlay.styles';

describe('<Overlay.Absolute />', () => {
  describe('component', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <div style={{ position: 'relative' }}>
          <Overlay.Absolute>Content</Overlay.Absolute>
        </div>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('css', () => {
    it('should match snapshot', () => {
      const css = absoluteCss({});

      expect(css).toMatchSnapshot();
    });

    it('should use provided options', () => {
      const css = absoluteCss({
        blur: 0.1,
        backgroundColor: 'a-100',
        backgroundOpacity: 0.1,
      });

      expect(css).toMatchSnapshot();
    });
  });
});

describe('<Overlay.FullScreen />', () => {
  describe('component', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <Overlay.FullScreen>Content</Overlay.FullScreen>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('css', () => {
    it('should match snapshot', () => {
      const css = fullScreenCss({});

      expect(css).toMatchSnapshot();
    });

    it('should use provided options', () => {
      const css = fullScreenCss({
        blur: 0.1,
        backgroundColor: 'a-100',
        backgroundOpacity: 0.1,
      });

      expect(css).toMatchSnapshot();
    });
  });
});
