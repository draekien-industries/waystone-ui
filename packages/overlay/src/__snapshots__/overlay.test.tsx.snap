// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay.Absolute /> component should match snapshot 1`] = `
<div>
  <div
    style="position: relative;"
  >
    <div
      class="css-7jeshx-AbsoluteOverlay"
    >
      Content
    </div>
  </div>
</div>
`;

exports[`<Overlay.Absolute /> css should match snapshot 1`] = `
{
  "backdropFilter": "blur(10px)",
  "backgroundColor": [Function],
  "borderRadius": "inherit",
  "inset": 0,
  "position": "absolute",
  "zIndex": "highest",
}
`;

exports[`<Overlay.Absolute /> css should use provided options 1`] = `
{
  "backdropFilter": "blur(0.1px)",
  "backgroundColor": [Function],
  "borderRadius": "inherit",
  "inset": 0,
  "position": "absolute",
  "zIndex": "highest",
}
`;

exports[`<Overlay.FullScreen /> component should match snapshot 1`] = `
<div>
  <div
    class="css-16wn25v-FullScreenOverlay"
  >
    <div
      class="css-1abarhv-FullScreenOverlay"
    >
      Content
    </div>
  </div>
</div>
`;

exports[`<Overlay.FullScreen /> css should match snapshot 1`] = `
{
  "backdropFilter": "blur(10px)",
  "backgroundColor": [Function],
  "inset": 0,
  "position": "fixed",
  "zIndex": "highest",
}
`;

exports[`<Overlay.FullScreen /> css should use provided options 1`] = `
{
  "backdropFilter": "blur(0.1px)",
  "backgroundColor": [Function],
  "inset": 0,
  "position": "fixed",
  "zIndex": "highest",
}
`;
